src/cpp/session/resources/schema/user-state-schema.json

```diff
@@ -4,6 +4,139 @@
     "title": "RStudio User State",
     "type": "object",
     "properties": {
+        "general": {
+            "type": "object",
+            "properties": {
+                "ignoredUpdateVersions": {
+                    "type": "array",
+                    "description": "A list of ignored updates. Ignored updates will not trigger an available update.",
+                    "items": {
+                        "type": "string"
+                    }
+                },
+                "clipboardMonitoring": {
+                    "type": "boolean",
+                    "description": "Linux clipboard monitoring for middle-click paste"
+                }
+            },
+            "default": {
+                "ignoredUpdateVersions": [],
+                "clipboardMonitoring": true
+            }
+        },
+        "font": {
+            "type": "object",
+            "properties": {
+                "proportionalFont": {
+                    "type": "string"
+                },
+                "fixedWidthFont": {
+                    "type": "string"
+                },
+                "useFontConfigDb": {
+                    "type": "boolean"
+                }
+            },
+            "description": "Font options from the Appearance category",
+            "default": {
+                "proportionalFont": "",
+                "fixedWidthFont": "",
+                "useFontConfigDb": true
+            }
+        },
+        "view": {
+            "type": "object",
+            "properties": {
+                "zoomLevel": {
+                    "type": "number",
+                    "description": "UI zoom level, 1.0 == 100%",
+                    "enum": [0.25, 0.5, 0.75, 0.8, 0.9, 1.0, 1.1, 1.25, 1.5, 1.75, 2.0, 2.5, 3.0, 4.0, 5.0]
```

Comment:
Is this a good opportunity to allow arbitrary zoom levels? (That is, we could just enforce that this is a number.)

Reply:
Or does it need to be this way to support the "legacy" options?

Reply:
Arbitrary zoom levels is possible and would be neat to have a zoom slider instead of the presets. It only has to be enforced for legacy options so I think it would be a requirement for Qt. Only validating for type number should be fine since the legacy value is not overwritten so a user could downgrade RStudio and go back to whatever settings they had.

Electron only needs a value greater than 0. Looks like Qt can only be 0.25 - 5.0. So I think keeping the minimum and maximum but allowing arbitrary values in between makes a lot of sense.